pkg/render/common/components/components.go

```diff
@@ -145,6 +162,29 @@ func ApplyDeploymentOverrides(d *appsv1.Deployment, overrides components.Replica
 	d.Spec.Strategy = *r.deploymentStrategy
 }
 
+// ApplyJobOverrides applies the overrides to the given Job.
+// Note: overrides must not be nil pointer.
+func ApplyJobOverrides(job *batchv1.Job, overrides components.PodResourceOverrides) {
+	// Catch if caller passes in an explicit nil.
```

Comment:
I know this is just copied from existing code, but the comments here are sort of wrong. The function level comment says "Must not be nil" and then the code actually handles `nil` explicitly, so seems to contradict the previous comment.